components/iceGrid.tsx

```diff
@@ -19,13 +23,22 @@ export function IceGrid({hexagonList, path}: IceGridProps) {
         <defs>
           <polygon id="hexagon" points="60,26 45,52 15,52 0,26 15,0 45,0"></polygon>
         </defs>
-        {hexagonList.map(([x, y]) => {
+        {hexagonList.map((hexagon) => {
+          const [x, y] = hexagon;
           const translateX = x * 46;
           const translateY = y * 54 + (isOdd(x) ? 27 : 0);
-          const isOnPath = path.find(([xPath, yPath]) => xPath === x && yPath === y);
+          const isOnPath = path.find((pathHexagon) => isEqual(hexagon, pathHexagon));
+          const isClickable = clickableHexagons.find((clickableHexagon) => isEqual(hexagon, clickableHexagon));
+          const isMistake = mistakeHexagons.find((mistakeHexagon) => isEqual(hexagon, mistakeHexagon));
 
           return (
-            <use className={isOnPath ? styles.hexagonPath : styles.hexagon} key={`${x},${y}`} xlinkHref="#hexagon" transform={`translate(${translateX},${translateY})`} />
+            <use 
+              className={isOnPath ? styles.hexagonPath : isMistake ? styles.hexagonMistake : isClickable ? styles.hexagonClickable : styles.hexagon} 
```

Comment:
There's a bunch of nested ternaries and repeated `find` code which we could probably simplify. Let's keep it in mind for some other time.